src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayOrderRequestBuilderTest.java

```diff
@@ -311,6 +312,67 @@ public void shouldGenerateValidAuthoriseGooglePay3DSOrderRequestWithoutIpAddress
         assertEquals(OrderRequestType.AUTHORISE_GOOGLE_PAY, actualRequest.getOrderRequestType());
     }
 
+    @Test
+    public void shouldGenerateValidAuthoriseGooglePay3DSOrderRequestWithIpAddress() throws Exception {
+        GooglePayAuthRequestFixture validGooglePay3dsData = anGooglePayAuthRequestFixture()
+                .withGooglePaymentInfo(new WalletPaymentInfo(
+                                "4242",
+                                "visa",
+                                PayersCardType.DEBIT,
+                                "Example Name",
+                                "example@test.example",
+                                GOOGLE_PAY_ACCEPT_HEADER,
+                                GOOGLE_PAY_USER_AGENT_HEADER,
+                                "8.8.8.8"
+                        )
+                );
+
+        GatewayOrder actualRequest = aWorldpayAuthoriseWalletOrderRequestBuilder(WalletType.GOOGLE_PAY)
+                .withWalletTemplateData(validGooglePay3dsData)
+                .with3dsRequired(true)
+                .withSessionId(WorldpayAuthoriseOrderSessionId.of("uniqueSessionId"))
+                .withAcceptHeader(GOOGLE_PAY_ACCEPT_HEADER)
+                .withUserAgentHeader(GOOGLE_PAY_USER_AGENT_HEADER)
+                .withPayerIpAddress("8.8.8.8")
+                .withTransactionId("MyUniqueTransactionId!")
+                .withMerchantCode("MERCHANTCODE")
+                .withDescription("This is the description")
+                .withAmount("500")
+                .build();
+
+        assertXMLEqual(TestTemplateResourceLoader.load(WORLDPAY_VALID_AUTHORISE_WORLDPAY_GOOGLE_PAY_3DS_REQUEST_WITH_IP_ADDRESS), actualRequest.getPayload());
+        assertEquals(OrderRequestType.AUTHORISE_GOOGLE_PAY, actualRequest.getOrderRequestType());
+    }
+
+    @Test
+    public void shouldGenerateValidAuthoriseGooglePay3DSOrderRequestWhen3dsDisabled() throws Exception {
```

Comment:
Can we use `3ds` instead of `3DS` for consistency?

```suggestion
    public void shouldGenerateValidAuthoriseGooglePay3dsOrderRequestWhen3dsIsDisabled() throws Exception {
```

Reply:
Done.